arcgis-ios-sdk-samples/Analysis/Analyze hotspots/AnalyzeHotspotsViewController.swift

```diff
@@ -99,6 +99,12 @@ class AnalyzeHotspotsViewController: UIViewController, HotspotSettingsVCDelegate
         }
     }
     
+    private func showAlert(error: Error) {
```

Comment:
I would recommend refactoring this into a convenience initializer of `UIAlertController`:
```swift
private extension UIAlertController {
    convenience init(error: Error) {
        self.init(errorMessage: error.localizedDescription)
    }
    
    convenience init(errorMessage message: String) {
        self.init(title: "Error", message: message, preferredStyle: .alert)
        let okayAction = UIAlertAction(title: "OK", style: .default)
        addAction(okayAction)
        preferredAction = okayAction
    }
}
```
You could then add a similar method to `UIViewController` via an extension:

```swift
private extension UIViewController {
    func showAlert(error: Error) {
        present(UIAlertController(error: error), animated: true)
    }
}
```

But I wouldn't recommend it for a couple reasons:

- The name `show` is similar to but semantically different from `UIViewController.show(_:sender:)`. "present" would be a better choice.
- The caller has no control over whether the presentation is animated.

Both of those issues could be resolved by renaming the method to something like `presentAlertController(for error: Error, animated: Bool)`, but at that point, it would be simpler for the view controller to just create and present the alert controller itself:

```
self.present(UIAlertController(error: error), animated: true)
```

Does that make sense?

Reply:
This makes sense but I'm not sure what the conclusion is. I renamed the `showAlert` functions to `presentAlert` to match the VC method; thanks for that suggestion. In any other app I would suggest making the extensions you mentioned, but in this case I think the simplest solution is to leave the code in private functions within each VC.

Reply:
My thought is that the code to create and present an alert controller doesn't add anything to the samples themselves. Putting that in a private extension at the bottom of the source file would keep it out of the way of the code that we actually want to show off.

Reply:
I agree. If we have an easy way to keep code that doesn't contribute to learning about the Runtime API in the context of the sample out of the way, then we should take it. Either a private extension or at least in a `Helper Methods` section at the very end of the class (as [here](https://github.com/Esri/arcgis-runtime-samples-ios/blob/8e18a910a7704ab34a86152a12f2502690c216fa/arcgis-ios-sdk-samples/Analysis/Statistical%20query%20(group%20and%20sort)/StatisticalQueryGroupAndSortViewController.swift#L393)).

I think in this case, since we're defining two simple methods over and over again (well, one and a convenience version of it), I would even be OK with having a single extension to `UIViewController` shared by all samples, but that might ruffle some feathers.

Reply:
Actually, having thought about this some more, I don't think it'll ruffle feathers. We'd be extending `UIViewController` and not one of the runtime classes. Sample code must show the Runtime APIs unmodified. Here we're showing an alert using code that we've written ourselves, so I would suggest a single implementation in a `UIViewController` extension that each sample can use.

Reply:
Changed to an extension in the latest commit.